tests/test_events_api.py

```diff
@@ -0,0 +1,166 @@
+from os import environ
+from typing import Callable
+from unittest.mock import MagicMock, patch
+
+import pytest
+
+from lbz.aws_boto3 import Boto3Client
+from lbz.events.api import BaseEvent, EventAPI
+
+
+class MyTestEvent(BaseEvent):
+    type = "MY_TEST_EVENT"
+
+
+class TestBaseEvent:
+    def test_base_event_creation_and_structure(self) -> None:
+        new_event = MyTestEvent({"x": 1})
+
+        assert hasattr(new_event, "raw_data")
+        assert hasattr(new_event, "data")
+        assert new_event.type == "MY_TEST_EVENT"
+        assert new_event.raw_data == {"x": 1}
+        assert new_event.data == '{"x": 1}'
+
+
+class TestEventApi:
+    def setup_method(self) -> None:
+        # pylint: disable= attribute-defined-outside-init
+        with patch.dict(environ, {"AWS_LAMBDA_FUNCTION_NAME": "million-dollar-lambda"}):
+            self.event_api = EventAPI()
+
+    def teardown_method(self, _test_method: Callable) -> None:
+        # pylint: disable= attribute-defined-outside-init
+        self.event_api._del()  # type: ignore # pylint: disable=protected-access
+
+    @patch.object(Boto3Client, "eventbridge", MagicMock())
+    def test___repr__(self) -> None:
+        expected_repr = (
+            "<EventAPI bus: million-dollar-lambda-event-bus Events: pending=0 sent=0 failed=0>"
+        )
+        assert str(self.event_api) == expected_repr
+        self.event_api.register(MyTestEvent({"x": 1}))
+        self.event_api.register(MyTestEvent({"x": 1}))
+        expected_repr = (
+            "<EventAPI bus: million-dollar-lambda-event-bus Events: pending=2 sent=0 failed=0>"
+        )
+        assert str(self.event_api) == expected_repr
+        self.event_api.send()
+        expected_repr = (
+            "<EventAPI bus: million-dollar-lambda-event-bus Events: pending=0 sent=2 failed=0>"
+        )
+        assert str(self.event_api) == expected_repr
+
+    @patch.object(Boto3Client, "eventbridge")
+    def test_settters(self, mock_send: MagicMock) -> None:
+        event = MyTestEvent({"x": 1})
+        self.event_api.register(event)
+        self.event_api.set_resources(["Yy", "ZZ"])
+        self.event_api.set_source("orgin")
+        self.event_api.set_bus_name("magic-bus")
+
+        self.event_api.send()
+
+        mock_send.put_events.assert_called_once_with(
+            Entries=[
+                {
+                    "Detail": '{"x": 1}',
+                    "DetailType": "MY_TEST_EVENT",
+                    "EventBusName": "magic-bus",
+                    "Resources": ["Yy", "ZZ"],
+                    "Source": "orgin",
+                }
+            ]
+        )
+
+    def test_register(self) -> None:
+        assert self.event_api.get_all_pending_events() == []
+
+        event = MyTestEvent({"x": 1})
+        self.event_api.register(event)
+
+        assert self.event_api.get_all_pending_events() == [event]
```

Comment:
This test is exactly the same as the below one but tests the single event - I think we can remove this one 😉